docs/README.md

```diff
@@ -88,7 +88,7 @@ Note: Other versions of roxygen2 might work in SparkR documentation generation b
 To generate API docs for any language, you'll need to install these libraries:
 
 ```sh
-pip install sphinx==2.3.1 mkdocs==1.0.4 numpy==1.18.1
+pip install -r ./dev/requirements.txt
```

Comment:
and TBH, I would like to don't pin the version here as well.

```diff
- pip install sphinx==2.3.1 mkdocs==1.0.4 numpy==1.18.1
+ pip install sphinx mkdocs numpy
```

Reply:
I'm confused. I thought you wanted to keep using a central `requirements.txt`, but you just wanted to not pin the versions.

Reply:
Yeah, I want. I want the central `requirements.txt` without pinning the version.

Reply:
Following from the comments by @cloud-fan [here](https://github.com/apache/spark/pull/27534#issuecomment-607011028) and [here](https://github.com/apache/spark/pull/28088#discussion_r401342910), and taking into account your request as well, I think we need to maintain two versions of the requirements file, one without pinning for dev purposes, and one with pinning for CI and the release machinery.

I will implement this using pip-tools, following the examples set by Warehouse and Trio which I referenced earlier in this PR.